src/bin/wasmer.rs

```diff
@@ -240,6 +251,49 @@ fn run(options: Run) {
     }
 }
 
+fn validate_wasm(validate: Validate) -> Result<(), String> {
+    let wasm_path = validate.path;
+    let wasm_path_as_str = wasm_path.to_str().unwrap();
+
+    let wasm_binary: Vec<u8> = read_file_contents(&wasm_path).map_err(|err| {
+        format!(
+            "Can't read the file {}: {}",
+            wasm_path.as_os_str().to_string_lossy(),
+            err
+        )
+    })?;
+
+    if !utils::is_wasm_binary(&wasm_binary) {
+        return Err(format!(
+            "Cannot recognize \"{}\" as a WASM binary",
+            wasm_path_as_str,
+        ));
+    }
+
+    wabt::Module::read_binary(wasm_binary, &Default::default())
```

Comment:
Runtime has a validate method:
https://github.com/wasmerio/wasmer/blob/bf823d2826aa2d17288103ba78bfc05a65e1f18c/lib/runtime/src/lib.rs#L87
It may be lighterweight solution than wabt.

Reply:
Hmm, I think we'd want better error messages than `yes` or `no`, I'll look in to the functions that that `validate` fn calls

Reply:
Thanks for pointing this out!